src/nwb_conversion_tools/tools/spikeinterface/spikeinterface.py

```diff
@@ -1132,35 +1133,40 @@ def add_units(
         )
         if property_name in ["max_channel", "max_electrode"] and nwbfile.electrodes is not None:
             unit_col_args.update(table=nwbfile.electrodes)
+
         units_table.add_column(**unit_col_args)
         aggregated_unit_properties[property_name] = checked_sorting.get_property(key=property_name)
-    for i, unit_id in enumerate(unit_ids):
-        if use_times:
-            spkt = checked_sorting.get_unit_spike_train(unit_id=unit_id, return_times=True)
-        else:
-            spkt = checked_sorting.get_unit_spike_train(unit_id=unit_id) / checked_sorting.get_sampling_frequency()
+
+    # Unit name logic
+    units_ids = checked_sorting.get_unit_ids()
+    if "unit_name" in aggregated_unit_properties:
+        unit_name_array = aggregated_unit_properties["unit_name"]
+    else:
+        unit_name_array = units_ids.astype("str", copy=False)
+        unit_col_args = dict(
+            name="unit_name",
+            description="unique reference for each unit",
+        )
+        units_table.add_column(**unit_col_args)
+        aggregated_unit_properties["unit_name"] = unit_name_array
+
+    for i, unit_id in enumerate(units_ids):
+        spkt = checked_sorting.get_unit_spike_train(unit_id=unit_id, return_times=True)
         kwargs = {key: val[i] for key, val in aggregated_unit_properties.items()}
 
-        units_table.add_unit(id=int(unit_id), spike_times=spkt, **kwargs)
+        # If the units ids are integer keep the old behavior of asigning table's id equal to unit_ids
+        if str(unit_id).isdigit():
+            id = int(unit_id)
+        else:
+            id = i
```

Comment:
Would it pose problems to let `add_unit` set the `id` by default?

Reply:
Compatibility with old behavior when the `unit_ids` are actually integers.

Reply:
So you're saying they wouldn't match the `unit_name` in that case?

Reply:
1) Let's suppose the unit_ids are 1, 3, 5, 8 (integers) 
2) Automatic assignation of id would created a unit table with ids 1, 2, 3, 4
3) The `NwbSortingExtractor` in `spikeextractors` gets its `units_ids` as 1, 2, 3, 4 and the comparison between the two of them will fail. 

This was happening with the Neuroscope recoding extractor if my memory does not fail me. I don't know if we have (or will have) a gin test with data like this.

Reply:
Ah, right